src/stores/ProfileController.ts

```diff
@@ -1,38 +1,42 @@
-import useService from '#src/hooks/useService';
+import { injectable } from 'inversify';
+
 import type { ProfilePayload, EnterProfilePayload, ProfileDetailsPayload } from '#types/account';
+import { ProfileService } from '#src/services/profile.service';
+import { useConfigStore } from '#src/stores/ConfigStore';
+
+@injectable()
+export default class ProfileController {
+  private readonly profileService: ProfileService;
+
+  constructor(profileService: ProfileService) {
+    this.profileService = profileService;
+  }
```

Comment:
I guess the ProfileService isn't a concrete injection? 

https://github.com/inversify/InversifyJS/blob/master/wiki/constructor_injection.md

Reply:
Is there a bug here? Maybe it needs to be optional (`?`)?It's possible, as the profile service was added after the fact and only came in when I merged latest from dev. I tried to refactor it quickly, but I might have missed something.

Reply:
No, I think it is correct like this. If the service was concrete, I would prefer using the concrete injection method so we can remove all optional chaining. But your other comment also answered this question 😄